src/app/api/core/utils/withRetry.ts

```diff
@@ -0,0 +1,49 @@
+import { StatusableError } from '@/types/CopilotApiError'
+import pRetry, { FailedAttemptError } from 'p-retry'
+import * as Sentry from '@sentry/nextjs'
+
+export const withRetry = async <T>(fn: (...args: any[]) => Promise<T>, args: any[]): Promise<T> => {
+  let isEventProcessorRegisterd = false
+
+  return await pRetry(
+    async () => {
+      try {
+        return await fn(...args)
+      } catch (error) {
+        // Hopefully now sentry doesn't report retry errors as well. We have enough triage issues as it is
+        Sentry.withScope((scope) => {
+          if (isEventProcessorRegisterd) return
+
+          isEventProcessorRegisterd = true
+          scope.addEventProcessor((event) => {
+            if (event.level === 'error' && event.message && event.message.includes('An error occurred during retry')) {
+              return null // Discard the event as it occured during retry
+            }
+            return event
+          })
+        })
+        // Rethrow the error so pRetry can rety
+        throw error
+      }
+    },
+
+    {
+      retries: 3,
+      minTimeout: 500,
+      maxTimeout: 500,
+      factor: 2, // Exponential factor for timeout delay. Tweak this if issues still persist
```

Comment:
The backoff logic with `factor: 2` is a good approach for retrying. Could you please help me understand the reason behind setting `maxTimeout` and `minTimeout` to 500?

Could setting both `minTimeout` and `maxTimeout` to the same value negate the exponential increase? I think we should consider allowing a larger `maxTimeout` (e.g., 2000) to provide actual exponential backoff benefits. What do you think?